R/save.r

```diff
@@ -158,18 +158,34 @@ plot_dev <- function(device, filename = NULL, dpi = 300) {
   force(dpi)
 
   if (is.function(device)) {
-    if ("file" %in% names(formals(device))) {
-      dev <- function(filename, ...) device(file = filename, ...)
-      return(dev)
-    } else {
-      return(device)
+    args <- formals(device)
+    call_args <- list()
+    if ("file" %in% names(args)) {
+      call_args$file <- filename
+    }
+    if ("res" %in% names(args)) {
+      call_args$res <- dpi
+    }
+    if ("units" %in% names(args)) {
+      call_args$units <- 'in'
```

Comment:
Is this correct? Shouldn't the `units` argument be handed through if the user provides a generic device function?

Reply:
Yes, this is correct. `ggsave()` converts the input and will always serve the dimensions as inches to the graphics device. This was the root cause of the whole ragg debacle

Reply:
Ah, I see.